### PR TITLE
Feat: Add Gladly connector

### LINKED ISSUE #935: Add Gladly connector

### PR DESCRIPTION
**Pull request by Beadko**
Closes #935 

## Checklist
- [x] Ran Linter
- [x] Created PR from non-main branch 

## Catalog variables
* {{workspace}}

## Notes
Pagination is not supported

**Media:**
Icon is in png format and has a white background
Logo is jpeg format and has a white background
No dark mode options are available

### GET 
URL: <localhost:4444/v1/agents> 
Screenshot 2567-08-18 at 14 36 50

### POST
URL: <localhost:4444/v1/answers> 
Screenshot 2567-08-18 at 16 14 05

### PUT 
URL: <localhost:4444/v1/answers/Pun0k4oxTA2CI_NeX4TixQ/languages/en-us/type/public> 
Screenshot 2567-08-18 at 16 18 58

### PATCH 
URL: <localhost:4444/v1/answers/VNEbRnbFR4Sb5YlWz5leuw> 
Screenshot 2567-08-18 at 16 15 14

### DELETE 
URL: <localhost:4444/v1/answers/VNEbRnbFR4Sb5YlWz5leuw> 
Screenshot 2567-08-18 at 16 16 03

## Pagination
Pagination is not supported
image
image
image

### CODE REVIEW

File: providers/gladly.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package providers
+
+const Gladly = "gladly"
+
+func init() {
+	SetInfo(Gladly, ProviderInfo{
+		DisplayName: "Gladly",
+		AuthType:    Basic,
+		BaseURL:     "https://{{.workspace}}/api",
```
Comment (jlimatampersand) [inline providers/gladly.go:9]: https://developer.gladly.com/rest/#section/Overview/REST-API

I think the base url is wrong. Please fix this.

**Review by jlimatampersand — CHANGES_REQUESTED** [review]

File: providers/gladly.go (hunk 1 above)
Comment (Beadko) [reply providers/gladly.go:9]: Withampersand's domain is partner-withampersand.us-uat.gladly.qa, when I tried to make API calls via partner-withampersand.us-uat.gladly.com, I get an error, when changing it to .gladly.qa, the call is successful. I have attached the screenshots
Screenshot 2567-08-20 at 15 40 27
Screenshot 2567-08-20 at 15 41 14

Comment (jlimatampersand) [reply providers/gladly.go:9]: I see where this is coming from thanks for the screenshot. But according to the doc https://developer.gladly.com/rest/#section/Overview/Testing

`.qa` is for testing sandbox environment and real accounts will be using `gladly.com`

I think the correct approach should be to use `gladly.com`. You can change this to `gladly.com` and I can accept this without testing `gladly.com` knowing the limitation. 

@laurenzlong what do you think? I think we can support QA testing environment later when we support modules.

Comment (jlimatampersand) [reply providers/gladly.go:9]: @Beadko as you change this to `gladly.com` you can add a comment in code that this does not support `.qa` yet.

Comment (jlimatampersand) [reply providers/gladly.go:9]: @Beadko Lauren and I had discussion on this, and decided that we should have separate connectors for qa environment and prod environment. Can you update this PR to have both connectors here?

**Review by jlimatampersand — APPROVED** [review]

**Pull request merged**